java/com/google/flatbuffers/Table.java

```diff
@@ -273,6 +284,320 @@ protected static int compareStrings(int offset_1, byte[] key, ByteBuffer bb) {
     }
     return len_1 - len_2;
   }
+
+  protected int __lookupByBooleanKey(int vectorFieldOffset, int keyFieldOffset, boolean queryKey,
+          boolean defaultValue ) {
+    byte key = (byte)(queryKey ? 1 : 0);
+    byte byteDefaultValue = (byte)(defaultValue ? 1 : 0);
+    return __lookupByByteKey(vectorFieldOffset, keyFieldOffset, key, byteDefaultValue);
+  }
+
+  protected int __lookupByByteKey(int vectorFieldOffset, int keyFieldOffset, int queryKey,
```

Comment:
By code bloat I meant execution time code bloat: if you have 5 object types with string key, then with generated code, the JIT has to work 5 times at optimizing those 5 lookup functions and generating the corresponding machine code. This put unwanted pressure on the processor instruction cache. Those issues are avoided by having a centralized implementation for a common type.

On another issue: How do you implement look-up by key using reflection if the look-up algorithm is only in the generated class code?